synapse/http/proxyagent.py

```diff
@@ -58,6 +59,9 @@ class ProxyAgent(_AgentBase):
 
         pool (HTTPConnectionPool|None): connection pool to be used. If None, a
             non-persistent pool instance will be created.
+
+        use_proxy (bool): Whether proxy settings should be discovered and used
+            from conventional environment variables. Defaults to false.
```

Comment:
Is there any point in using `ProxyAgent` without `use_proxy=True`? I kept this default here to match the same behaviour as before if you didn't specify either of `http_proxy` or `https_proxy` but seems like it may be redundant now that this is combined into a single parameter

Reply:
I think you're right. Outside of proxying (and a single debug log) `ProxyAgent` doesn't provide much on top of twisted's default `Agent` (as it shouldn't!).

I think we could try just using an `Agent` if `use_proxy=False` is passed to `SimpleHttpClient`.

Similarly in tests.

Reply:
I think I'd rather leave this out of this PR for now to avoid making more significant changes to the code but sounds like it could be worth a follow up!

Reply:
Sounds sensible, OK :slightly_smiling_face: